lib/qesapdeployment.pm

```diff
@@ -91,7 +102,103 @@ sub qesap_get_deployment_code {
         push(@log_files, QESAP_GIT_CLONE_LOG);
         assert_script_run("set -o pipefail ; $git_clone_cmd | tee " . QESAP_GIT_CLONE_LOG, quiet => 1);
     }
+    assert_script_run("ln -s " . TERRAFORM_DIR . "/aws " . TERRAFORM_DIR . "/ec2");
+}
+
+
+=head3 qesap_upload_logs
+
+    Collect and upload logs present in @log_files.
+=cut
+
+sub qesap_upload_logs {
+    record_info("Uploading logfiles", join("\n", @log_files));
+    for my $file (@log_files) {
+        upload_logs($file, failok => 1);
+    }
+    # Remove uploaded files from arrays
+    @log_files = ();
+}
+
+
+=head3 qesap_yaml_replace
+
+    Replaces yaml config file variables with the ones defined in openqa.
+    Openqa variables need to be added as 'openqa_variables' inside %args as following:
+    <OPENQA_VARIABLE_UPPERCASE> = "<VARIABLE_VALUE>"
+
+=cut
+
+sub qesap_yaml_replace {
+    my (%run_args) = @_;
+    my %variables = $run_args{openqa_variables};
```

Comment:
Isn't it better to pass a list of settings to replace and iterate with `get_var()` instead of duplicating these that are already defined in `%bmwqemu::vars`? I'd like to understand why they are needed in `%run_args` since all vars should be available to all test modules all the time.

Reply:
>  iterate with get_var() instead 

not necessary all of them come from get_var

Reply:
Yes, but it's duplicating memory without it being necessary AFAICS. All settings are in memory, available to all modules in  `%bmwqemu::vars`, then there some of these are duplicated in `%run_args{openqa_variables}` after calling `qesap_get_variables()`, and this copy it's the one used here. The question is why do we need to duplicate these, when the ones in `%bmwqemu::vars` are accessible to all modules via `get_var()`. If the duplication is introduced only to have a subset with which to iterate in the code of `qesap_yaml_replace()`, then the same can be accomplished with a list of the settings to replace and `get_var()`

Reply:
> Isn't it better to pass a list of settings to replace and iterate with `get_var()` instead of duplicating these that are already defined in `%bmwqemu::vars`? I'd like to understand why they are needed in `%run_args` since all vars should be available to all test modules all the time.

Yeah I have done it like that after suggestion from @mpagot that the variables should be handled by testcode for each project separately.
With passing only a list there is a problem with defining default values like:

```
    $variables{FENCING_MECHANISM} = get_var("FENCING_MECHANISM", "sbd");
```

Not the best example but I saw in the old code parameters defined like:

```
my $sle_version = get_var('FORCED_DEPLOY_REPO_VERSION') ? get_var('FORCED_DEPLOY_REPO_VERSION') : get_var('VERSION');
```
But the point is not to be limited only by output form get_var()

Reply:
Is it feasible to have an interface like:

```
# Here we are in th etest code
my variables_for_this_test = here a map with all needed variables and values
qesap_yaml_replace(%variables_for_this_test);
```

Reply:
I think this is what Lumir is providing with `qesap_get_variables()`

Reply:
Correct. Everything needed is just to define variables in $run_args{openqa_variables}.
I have added an example how to use it as well.